package.json

```diff
@@ -42,7 +42,8 @@
     "prettier": "prettier --write \"**/*.{js,ts,tsx,md}\"",
     "prettier:check": "prettier --check \"**/*.{js,ts,tsx,md}\"",
     "test": "jest --env jsdom",
-    "types": "tsc --project tsconfig.json",
+    "emit-types": "lerna run emit-types",
```

Comment:
This can be done in parallel with building components right? I think it's nice to have it as its own method too, but we can optimize the actual build step.

Reply:
Given we use `CleanWebpackPlugin`, the types should probably run after, not in parallel, to avoid a race condition issue.

Reply:
That plugin causes so many troubles. We should have a separate command of `clean` that runs `npx lerna clean` and removing the `dist` folders.

Reply:
It's been fine since we made our packages externals, and I rather like deferring to webpack cleaning up before emitting bundles. Ideally, I think webpack would deal with outputting the definition files as well.

Is it actually a problem for us to have these two commands run synchronously? It's not part of the development workflow, only the distribution flow.

A clean command seems nice and we may want one, but I think that doing `npm run clean && npm-run-all --parallel build ts:emit` would likely take longer overall to run despite the two build commands running in parallel. Could test that theory, of course.

Reply:
`npm run build` is part of the developer setup workflow. That's the main reason I raised the objection.

This could be part of `prepublishOnly` hook instead. I'd be fine to have it run sync after build to keep the two commands separate.

Reply:
That's true. If we only have this run on prepublish though, we don't get the benefits of the definition files when we're working with packages internally like lib.